extensions/amp-facebook-comments/0.1/amp-facebook-comments.js

```diff
@@ -69,11 +76,39 @@ class AmpFacebookComments extends AMP.BaseElement {
     listenFor(iframe, 'embed-size', data => {
       this./*OK*/changeHeight(data['height']);
     }, /* opt_is3P */true);
+    this.unlistenMessage_ = listen(
+        this.win,
+        'message',
+        this.handleFacebookMessages_.bind(this)
+    );
+    this.toggleLoading(true);
     this.element.appendChild(iframe);
     this.iframe_ = iframe;
     return this.loadPromise(iframe);
   }
 
+  /** @private */
+  handleFacebookMessages_(event) {
+    if (this.iframe_ && event.source != this.iframe_.contentWindow) {
+      return;
+    }
+    if (!getData(event) || !(isObject(getData(event))
+        || startsWith(/** @type {string} */ (getData(event)), '{'))) {
+      return; // Doesn't look like JSON.
+    }
+
+    /** @const {?JsonObject} */
+    const eventData = /** @type {?JsonObject} */ (isObject(getData(event))
```

Comment:
When would `event.data` be a JSON string and when would it be a JSON object? Does the Facebook API return both?

Reply:
No this is because the iframe is also sending some ill formed JSON messages which I need to gracefully handle as well. Had to do the same for `amp-bodymovin-animation.js` - https://github.com/ampproject/amphtml/blob/master/extensions/amp-bodymovin-animation/0.1/amp-bodymovin-animation.js#L145.